### PR TITLE
[SN-788] Browse filters v1

### PR DESCRIPTION
**Pull request by jasont7**
This PR introduces:
- Brands filter
- Size filter (men's and women's sizes)
- Sort by options (relevance, trending, newest, price low, price high)

Note: these filters can also be combined with search querying.

### CODE REVIEW

File: src/assets/constants.js [hunk 1]
```diff
@@ -96,74 +104,26 @@ export const websiteLogoMapGrey = {
...
-
-// browse - multi-select filter options
-function generateReleaseYears () {
-  const currentYear = new Date().getFullYear()
-  const years = []
-  for (let i = currentYear; i >= 2000; i--) {
-    years.push({ value: i.toString(), label: i.toString() })
-  }
-  return years
+export const sortByOptionsFormattedMap = {
```
Comment (EricHasegawa) [inline src/assets/constants.js:121]: Change name to `sortByOptionToDisplayText`? It's same # of chars but I think more descriptive

File: src/assets/styling/_sortButton.scss [hunk 2]
```diff
@@ -57,8 +57,9 @@
 }
 
 @media screen and (max-width: 547px) {
-  .category-filter {
+  .sort-button {
     min-width: 335px;
     margin-left: 0px;
   }
 }
+
```
Comment (EricHasegawa) [inline src/assets/styling/_sortButton.scss:65]: image
no newline?

File: src/assets/styling/_sortModal.scss [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        .sort-modal-buttons {
+          grid-template-columns: repeat(1, 1fr);
+          width: 280px;
+          padding-top: 17px;
+        }
+      }
+    }
+  }
+}
+
```
Comment (EricHasegawa) [inline src/assets/styling/_sortModal.scss:117]: again with the newline

File: src/components/Browse/brandOption.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { toggleBrowseBrand } from '../../redux/actions'
+import PropTypes from 'prop-types'
+
+export default function BrandOption ({ brand }) {
+  const dispatch = useDispatch()
+
+  const brands = useSelector(state => state.browse.filters.brands)
```
Comment (EricHasegawa) [inline src/components/Browse/brandOption.js:9]: I think we should revisit how many global state variables we have at some point, its probably better to limit some of these in scope (that would be for another PR)

File: src/components/Browse/sortByOption.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+export default function SortByOption ({ sortBy }) {
+  const dispatch = useDispatch()
+
+  const sortByState = useSelector(state => state.browse.filters.sortBy)
+
+  const clickHandler = () => {
+    dispatch(updateBrowseSort(sortBy))
+  }
+
+  const sortByOptionClassName = sortByState === sortBy
```
Comment (EricHasegawa) [inline src/components/Browse/sortByOption.js:16]: You can move this directly into line 21 (I think it's better that way but kinda a personal preference thing)

File: src/components/Modals/sizeModal.js [hunk 6]
```diff
@@ -28,7 +34,7 @@ export default function SizeModal ({ gender }) {
               <Close />
             </button>
           </div>
-          d
```
Comment (EricHasegawa) [inline src/components/Modals/sizeModal.js:31]: i thought `d` was a nice touch

File: src/hooks/createRequest.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 
 export default function createRequestObject (type, filter) {
-  const api = 'https://rkzmvt26dg.us-west-2.awsapprunner.com'
-  // const api = 'http://127.0.0.1:8080'
+  // const api = 'https://rkzmvt26dg.us-west-2.awsapprunner.com'
+  const api = 'http://127.0.0.1:8086'
```
Comment (EricHasegawa) [inline src/hooks/createRequest.js:4]: this is not a senior staff principal swe move bro

File: src/hooks/scrapers.js [hunk 8]
```diff
@@ -4,10 +4,10 @@ import createRequestObject from './createRequest'
 
 export async function stockxLowestPrice (item, filter) {
   try {
-    const modelName = item.modelName.replace('(W)', '')
+    const model = item.model.replace('(W)', '')
```
Comment (EricHasegawa) [inline src/hooks/scrapers.js:7]: Can we do this model and sku formatting once in `getItemPrices` then just pass down the modified item? Feel like its better than having it done all over.

**Review by EricHasegawa — CHANGES_REQUESTED** [review]
Looks great - mostly minor changes, testing locally now

File: src/components/Browse/brandOption.js (hunk 4 above)
Comment (jasont7) [reply src/components/Browse/brandOption.js:9]: I think I agree when it comes to singleton states (like size and currency), but here the `brands` state is part of the larger filters state which itself is part of the browse state so it's valid imo

File: src/hooks/scrapers.js (hunk 8 above)
Comment (jasont7) [reply src/hooks/scrapers.js:7]: will be addressed when transferring input formatting to backend

**Review by EricHasegawa — APPROVED** [review]
